File: events/leave.py
import logging
from slack_sdk import WebClient
from events.blacklist import blockIfBlacklisted

logger = logging.getLogger(__name__)

def register(app):
    @app.command("/yapperleave")
    @blockIfBlacklisted
    def leaveChannel(ack, body, client: WebClient, respond):
        ack()
        channelId = body["channel_id"]
        try:
            client.chat_postMessage(channel=channelId, text=f"👋 Leaving <#{channelId}> now. Bye!")
            client.conversations_leave(channel=channelId)

            msg = f"Bot left channel {channelId}"
            logger.info(f"[leave] {msg}")

        except Exception as e:
            errorMsg = f"Failed to leave channel {channelId}: {e}"
            logger.error(f"[leave] {errorMsg}")
            respond(f"⚠️ Error: {e}")

File: events/leave.js
import { withBlacklistCommand } from './blacklist.js';

export function register(app) {
  app.command("/yapperleave", withBlacklistCommand('yapperleave', async ({ ack, body, client, respond }) => {
    await ack();
    const channelId = body.channel_id;

    try {
      await client.chat.postMessage({
        channel: channelId,
        text: `👋 Leaving <#${channelId}> now. Bye!`
      });

      await client.conversations.leave({ channel: channelId });

      console.log(`[leave] Bot left channel ${channelId}`);
    } catch (e) {
      console.error(`[leave] Failed to leave channel ${channelId}: ${e}`);
      await respond(`⚠️ Error: ${e}`);
    }
  }));
}